tket2-hseries/src/lazify_measure.rs

```diff
@@ -74,158 +82,244 @@ impl LazifyMeasurePass {
+            let mut v = hugr.linked_inputs(node, 1).collect_vec();
+            v.sort();
+            v
+        };
+        let (lazy_measure_reset_node, replacement) = {
+            let bool_uses = uses.len();
+            let mut builder = {
+                let outputs = iter::once(qb_t())
+                    .chain(itertools::repeat_n(bool_t(), bool_uses))
+                    .collect_vec();
+                DFGBuilder::new(Signature::new(qb_t(), outputs)).unwrap()
+            };
+            let [qb] = builder.input_wires_arr();
+            let (lazy_measure_reset_node, [qb_wire, future_wire]) = {
+                let handle = builder
+                    .add_dataflow_op(QSystemOp::LazyMeasureReset, [qb])
+                    .unwrap();
+                (handle.node(), handle.outputs_arr())
+            };
+            let out_wires = Self::build_futures_gadget(&mut builder, future_wire, bool_uses);
+            (
+                lazy_measure_reset_node,
+                builder
+                    .finish_hugr_with_outputs(iter::once(qb_wire).chain(out_wires))
+                    .unwrap(),
+            )
+        };
+        let nu_inp = HashMap::from_iter([((lazy_measure_reset_node, 0.into()), (node, 0.into()))]);
+        let qb_use = hugr.single_linked_input(node, 0).unwrap(); // qubit is linear so this can't fail
+        let nu_out = iter::zip(iter::once(qb_use).chain(uses), (0..).map_into()).collect();
 
-fn simple_replace_measure(
-    hugr: &impl HugrView,
-    node: Node,
-) -> (HashSet<(Node, IncomingPort)>, SimpleReplacement) {
-    assert!(
-        hugr.get_optype(node).cast() == Some(Tk2Op::Measure),
-        "{:?}",
-        hugr.get_optype(node)
-    );
-    let g = SiblingSubgraph::try_from_nodes([node], hugr).unwrap();
-    let num_uses_of_bool = hugr.linked_inputs(node, OutgoingPort::from(1)).count();
-    let replacement_hugr = measure_replacement(num_uses_of_bool);
-    let [i, o] = replacement_hugr.get_io(replacement_hugr.root()).unwrap();
+        Ok(Self(SimpleReplacement::new(
+            subgraph,
+            replacement,
+            nu_inp,
+            nu_out,
+        )))
+    }
 
-    // A map from (target ports of edges from the Input node of `replacement`) to (target ports of
-    // edges from nodes not in `removal` to nodes in `removal`).
-    let nu_inp = replacement_hugr
-        .all_linked_inputs(i)
-        .map(|(n, p)| ((n, p), (node, p)))
-        .collect();
+    fn build_futures_gadget(builder: &mut impl Dataflow, wire: Wire, num_uses: usize) -> Vec<Wire> {
+        let future_wires = if num_uses == 0 {
+            builder.add_free(wire, bool_t()).unwrap();
+            vec![]
+        } else {
+            let mut future_wires = vec![wire];
+            for _ in 1..num_uses {
+                let prev_wire = future_wires.last_mut().unwrap();
+                let [wire1, wire2] = builder.add_dup(*prev_wire, bool_t()).unwrap();
+                *prev_wire = wire1;
```

Comment:
does this line actually do anything? `prev_wire` doesn't seem to be used after this line